tools/static-assets/server/boot.js

```diff
@@ -492,6 +492,14 @@ var runMain = Profile("Run main()", async function () {
   }
 });
 
+
+// As our main process runs now inside a Promise, we need to make that if the user has an unhandled promise
+// in their app, ths unhandledRejection doesn't crash the server
+process.on("unhandledRejection", (e) =>
+  console.error("Unhandled promise rejection:", e.stack || e)
+);
+
+
```

Comment:
I like this, but this might be a workaround.

Reply:
Whenever an "unhandledRejection" happens, it's crucial not only to log it but also to address and handle the failing promises. I wonder if we just ignore the crashing this could affect also the platform code itself and we potentially can have false positive during tests,  without our awareness.

Maybe we could handle it by logging it and give some extra instructions on ideas for devs to act on it, but crash it anyway. In the instructions we may include [@zodern](https://github.com/meteor/meteor/pull/13062#issuecomment-2004117304) suggestion of adding the option `--unhandled-rejections` on run the app and link to node's api to properly handle `Promise.rejection`.

Reply:
> Maybe we could handle it by logging it and give some extra instructions

Good idea! Let's discuss this later and open it in other PR.